enhancements/network/static-ip-addresses-vsphere.md

```diff
@@ -0,0 +1,429 @@
+      failureDomain: us-east-1c
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.243/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: compute
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.244/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: compute
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.245/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: compute
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.246/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+~~~
+2. Add validation for the modified/added fields in the platform specification.
+3. For compute nodes, produce machine manifests with associated network device configuration.  
+
+Example of `machine` configured with network device configuration
+~~~yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: Machine
+metadata:
+  name: test-compute-1
+spec:
+  lifecycleHooks: {}
+  metadata: {}
+  providerSpec:
+    value:
+      numCoresPerSocket: 2
+      diskGiB: 60
+      snapshot: ''
+      userDataSecret:
+        name: worker-user-data
+      memoryMiB: 8192
+      credentialsSecret:
+        name: vsphere-cloud-credentials
+      network:
+        devices:
+          - networkName: lab
+            ipAddrs:
+            - 192.168.101.244/24
+            gateway4: 192.168.101.1
+            nameservers:
+            - 192.168.101.2            
+      metadata:
+        creationTimestamp: null
+      numCPUs: 2      
+      kind: VSphereMachineProviderSpec
+      workspace:
+        datacenter: testdc
+        datastore: datastore-1
+        folder: /testdc/vm/cluster-folder
+        resourcePool: /testdc/host/cluster1/Resources
+        server: test.vcenter.net
+      template: vm-template-rhcos
+      apiVersion: machine.openshift.io/v1beta1
+~~~
+4. For bootstrap and control plane nodes, modify vSphere terraform to convert network device configuration to a VM guestinfo parameter
+for each VM to be created.
+
+As the assets are generated for the control plane and compute nodes, the slice of `host`s for each 
+node role will be used to populate network device configuration.  The number of `host`s must match the number of
+replicas defined in the associated machine pool.
+
+Additionally, each defined host may optionally define a failure domain.  This indicates that the associated `networkDevice` will be applied to a machine created in the indicated failure domain.
+
+
+##### Machine API
+- Modify vSphere machine controller to convert IP configuration to VM guestinfo parameter
+- Introduce a new lifecycle hook called `preProvision`.
+- Modify [types_vsphereprovider.go](https://github.com/openshift/api/blob/master/machine/v1beta1/types_vsphereprovider.go) to support network device configuration. 
+
+
+###### network device configuration of Machines
```

Comment:
At present, the installer doesn't set any fields specific to static IPs for the control plane nodes.  Rather, terraform passes the static IP along in the network kargs.  The CPMS manifest could be modified as a day 2 operation to define an `ippool` or we could require an IPAM controller to be present during installation, but this would complicate the installation of the cluster.  The thought I was working towards was to enable the installer to, on its own, provision a minimal cluster with the IPAM controller being used to provision static IPs as a day 2 operation.

Reply:
I believe we have this one covered now with @vr4manta's update.